databricks/koala/namespace.py

```diff
@@ -123,3 +126,59 @@ def _to_datetime2(arg_year, arg_month, arg_day,
         errors=errors,
         format=format,
         infer_datetime_format=infer_datetime_format)
+
+
+@derived_from(pd)
+def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False, columns=None, sparse=False,
+                drop_first=False, dtype=None):
+    if sparse is not False:
+        raise NotImplementedError("get_dummies currently does not support sparse")
+
+    if isinstance(columns, string_types):
+        columns = [columns]
+    if dtype is None:
+        dtype = 'byte'
+
+    if isinstance(data, Column):
+        if prefix is not None:
+            prefix = [str(prefix)]
+        columns = [data.name]
+        df = data.to_dataframe()
+        remaining_columns = []
+    else:
+        if isinstance(prefix, string_types):
+            raise ValueError("get_dummies currently does not support prefix as string types")
+        df = data.copy()
+        if columns is None:
+            columns = [column for column in df.columns
+                       if not isinstance(data.schema[column].dataType, NumericType)]
+        if prefix is None:
+            prefix = columns
+        remaining_columns = [df[column] for column in df.columns if column not in columns]
```

Comment:
columns is a list so that makes an O(N^2) algorithm. Let's take all the column names into a set, just in case someone has 1000 columns.